flight/PiOS/inc/pios_mpu.h

```diff
@@ -0,0 +1,192 @@
+/**
+ ******************************************************************************
+ * @file       pios_mpu_priv.h
+ * @author     dRonin, http://dRonin.org/, Copyright (C) 2015
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2012-2015
+ * @author     The OpenPilot Team, http://www.openpilot.org Copyright (C) 2012.
+ * @addtogroup PIOS PIOS Core hardware abstraction layer
+ * @{
+ * @addtogroup PIOS_MPU Invensense MPU Functions
+ * @{
+ * @brief Common driver for Invensense MPU-xxx0 6/9-axis IMUs.
+ * Public Interface
+ *****************************************************************************/
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#ifndef PIOS_MPU_H
+#define PIOS_MPU_H
+
+#include "pios.h"
+
+/**
+ * MPU driver error return types
+ */
+enum pios_mpu_error {
+	PIOS_MPU_ERROR_NONE     = 0,
+	PIOS_MPU_ERROR_NOCONFIG = 1,
+	PIOS_MPU_ERROR_PROBEFAILED,
+	PIOS_MPU_ERROR_WRITEFAILED,
+	PIOS_MPU_ERROR_READFAILED,
+	PIOS_MPU_ERROR_NOIRQ,
+	PIOS_MPU_ERROR_SAMPLERATE,
+	PIOS_MPU_ERROR_WHOAMI,
+	PIOS_MPU_ERROR_MAGFAILED,
+};
+
+/**
+ * Supported IMU devices
+ */
+enum pios_mpu_type {
+	PIOS_MPU60X0, /**< MPU-6000, SPI/I2C or MPU-6050, I2C */
+	PIOS_MPU6500, /**< MPU-6500, SPI/I2C */
+	PIOS_MPU9150, /**< MPU-9150, contains MPU-6050+AK8975C */
+	PIOS_MPU9250, /**< MPU-9250, contains MPU-6500+AK8963 */
+	PIOS_MPU_NUM  /**< Number of elements */
+};
+
+enum pios_mpu60x0_gyro_filter {
+	PIOS_MPU60X0_GYRO_LOWPASS_256_HZ = 0x00,
+	PIOS_MPU60X0_GYRO_LOWPASS_188_HZ = 0x01,
+	PIOS_MPU60X0_GYRO_LOWPASS_98_HZ  = 0x02,
+	PIOS_MPU60X0_GYRO_LOWPASS_42_HZ  = 0x03,
+	PIOS_MPU60X0_GYRO_LOWPASS_20_HZ  = 0x04,
+	PIOS_MPU60X0_GYRO_LOWPASS_10_HZ  = 0x05,
+	PIOS_MPU60X0_GYRO_LOWPASS_5_HZ   = 0x06
+};
+
+enum pios_mpu6500_gyro_filter {
+	PIOS_MPU6500_GYRO_LOWPASS_250_HZ = 0x00,
+	PIOS_MPU6500_GYRO_LOWPASS_184_HZ = 0x01,
+	PIOS_MPU6500_GYRO_LOWPASS_92_HZ  = 0x02,
+	PIOS_MPU6500_GYRO_LOWPASS_41_HZ  = 0x03,
+	PIOS_MPU6500_GYRO_LOWPASS_20_HZ  = 0x04,
+	PIOS_MPU6500_GYRO_LOWPASS_10_HZ  = 0x05,
+	PIOS_MPU6500_GYRO_LOWPASS_5_HZ   = 0x06
```

Comment:
Not going to do it for these ones since they're locked into silicon/won't change.